x-pack/plugins/ml/server/models/data_recognizer/modules/apm_transaction/ml/apm_metrics.json

```diff
@@ -0,0 +1,53 @@
+{
+  "job_type": "anomaly_detector",
+  "groups": [
+    "apm"
+  ],
+  "description": "Detects anomalies in transaction duration, throughput and error percentage for metric data.",
+  "analysis_config": {
+    "bucket_span": "15m",
+    "summary_count_field_name" : "doc_count",
+    "detectors" : [
+      {
+        "detector_description" : "high duration by transaction type for an APM service",
+        "function" : "high_mean",
+        "field_name" : "transaction.duration.us",
```

Comment:
Why is the detector field name different from the field that is aggregated on (in the datafeed config)?

Reply:
Ah, it's not a field name, but an aggregation name. Can we rename it to something that isn't an Elasticsearch field name? or are there any reasons why it should be the same?

Reply:
@dgieselaar when querying the resulting data, it might be good to have a 1-1 field name mapping. So you can query the true value of `transaction.duration.histogram` and aggregate it directly with the resulting model-plot document values.

Reply:
`transaction.duration.us` shows up in `field_name` as a value, not as an ES field name, correct?